### ISSUE TITLE
metadata retrieval requires ids not Into<ids>

### ISSUE THREAD

**Issue by molpopgen** [tskit-dev/tskit-rust#249]
For example, from `edge_table.rs`:

```rust
    pub fn metadata<T: metadata::MetadataRoundtrip>(
        &'a self,
        row: EdgeId,
    ) -> Result<Option<T>, TskitError> {
        let buffer = metadata_to_vector!(self, row.0)?;
        decode_metadata_row!(T, buffer)
    }
```

The type of row should be `E: Into<EdgeId>`

**Comment by molpopgen** [comment]
I think there's a simpler solution.
The argument can be `impl Into<T>`
instead of a manually-named generic.

**Comment by molpopgen** [comment]
1.63.0 allows what we want via a backwards-compatible change.

**Comment by molpopgen** [comment]
> 1.63.0 allows what we want via a backwards-compatible change.

Since I didn't write a note about what this was:

`fn foo(x: impl Into<Y>)` is what I meant.

**Comment by molpopgen** [comment]
closed via #633
